lib/tapioca/compilers/requires_compiler.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+# typed: true
+
+require_relative '../sorbet_config_parser'
+
+module Tapioca
+  module Compilers
+    class RequiresCompiler
+      extend T::Sig
+
+      sig { params(sorbet_path: String).void }
+      def initialize(sorbet_path)
+        @sorbet_path = sorbet_path
+      end
+
+      sig { returns(String) }
+      def compile
+        config = SorbetConfig.parse_file(@sorbet_path)
+        files = collect_files(config)
+        files.flat_map do |file|
+          collect_requires(file).reject do |req|
+            name = collect_required_name(T.cast(req, String))
+            next unless name
+            name_in_project?(files, T.must(name[1]))
+          end
+        end.sort.uniq.join
+      end
+
+      private
+
+      sig { params(config: SorbetConfig).returns(T::Array[String]) }
+      def collect_files(config)
+        config.paths.flat_map do |path|
+          path = (Pathname.new(@sorbet_path) / "../.." / path).cleanpath
+          if path.directory?
+            Dir.glob("#{path}/**/*.rb", File::FNM_EXTGLOB).reject do |file|
+              file_ignored_by_sorbet?(config, file)
+            end
+          else
+            [path.to_s]
+          end
+        end.sort.uniq
+      end
+
+      sig { params(file_path: String).returns(T::Enumerable[String]) }
+      def collect_requires(file_path)
+        File.read(file_path).lines.select do |line|
+          line =~ /^require ?\(?['"]/
```

Comment:
I think this is still a little too restrictive since it won't match `require`s that are indented like `  require "foo"`.

Reply:
The goal with this heuristic is to avoid all the requires from inside the code, for example the conditional requires depending on platform, etc. as to minimize the amount of gems you need to remove manually because `tapioca` cannot load them.

I can go  and grab all of them and let the user remove them if you prefer.

Reply:
Hmm, not sure how I feel about this one. Conditional requires would still imply that we would need them for generating RBIs, and, in some cases, maybe even more so. There are also patterns like:
```ruby
begin
  require "foo"
rescue NameError
  require "foo_fallback"
end
```
that is used in some apps, so might be good to catch those too.

Do you have any examples of matching indented requires resulting in a lot of extra requires?

Reply:
Ok, I selected all `require`s whatever their indentation level.